mantidproject/mantid
# Stop ws locks causing context menus to wait

The workspace context menu was getting stuck on using an algorithm HasUB to determine weether to show a "Clear UB" option.

These changes add a direct HasOrientedLattice to all IMDWorkspaces and anything that inherits from it, so you don't need to use the algorithm and get stopped by the ws locking.

**To test:**
To test this you need to run something slow like:
``` python
CreateSampleWorkspace(OutputWorkspace='base', Function='Powder Diffraction', NumBanks=50, BankPixelWidth=100)
```
1. Run the alg to create a workspace, right click on the workspace and see the context menu
2. Rerun the same algorithm with the same output ws (this will cause the ws write lock)
3. While the algorithm is running right click on the workspace and get the context menu up (it might be a bit sluggish, but should appear.


Fixes #21263 

**Release Notes** 

docs/source/release/... /framework.rst


#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
